tsconfig.json

```diff
@@ -1,10 +1,36 @@
 {
+  // see https://www.typescriptlang.org/tsconfig to better understand tsconfigs
+  "include": ["src", "types"],
   "compilerOptions": {
+    "module": "esnext",
+    "lib": ["dom", "esnext"],
+    "importHelpers": true,
+    // output .d.ts declaration files for consumers
     "declaration": true,
-    "lib": ["DOM", "ESNext"],
-    "module": "ESNext",
+    // output .js.map sourcemap files for consumers
+    "sourceMap": true,
+    // match output dir to input dir. e.g. dist/index instead of dist/src/index
+    "rootDir": "./src",
+    // stricter type-checking for stronger correctness. Recommended by TS
     "strict": true,
-    "target": "ES6"
-  },
-  "files": ["lib/pin-field.ts"]
+    // linter checks for common issues
+    "noImplicitReturns": true,
+    "noFallthroughCasesInSwitch": true,
```

Comment:
What are your motivations for this one?

Reply:
Good you are taking this on for your needs, like discussed. I did just copy the basic tsconfig from tsdx - and did minimal changes for it to look ok.

cheers